File: src/ProjectR.Scripting/content/scripts/spells/lua/BreathOfTheBabyIceDragon.lua
function GetTargetType()
    return Single
end

function GetName()
return "Breath of the Baby Ice Dragon"
end

function GetDescription()
    return "A breath attack using both your AD & MD to slow your target."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .2
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Composite
end

function GetMasteries()
  return { ICE }
end

function SpellEffect()
  local damage = (((2.53 * aAD + 2.53 * aMD) * (aICE/100.)) - (.8 * dDEF + .8 * dMR)) * (100/dICE)

  d:BuffStat(SPD, -.5)

  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/BreathOfTheBabyIceDragon.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class BreathOfTheBabyIceDragon : SpellScriptBase
    {
        public override string Name { get { return "Breath of the Baby Ice Dragon"; } }
        public override string Description { get { return "A breath attack using both your AD & MD to slow your target."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ICE,  }; } }
        public override SpellType SpellType { get { return SpellType.Composite; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 20; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (((2.53 * cAD + 2.53 * cMD) * (cICE/100)) - (.8 * tDEF + .8 * tMR)) * (100/tICE);

            BuffStat(Stat.SPD, -.5);

            DealDamage(damage);
        }
    }
}